src/main/java/com/conveyal/gtfs/PatternFinder.java

```diff
@@ -260,8 +265,8 @@ private static String getTerminusName(
             return stopById.get(haltId).stop_name;
         } else if (locationById.containsKey(haltId)) {
             return locationById.get(haltId).stop_name;
-        } else if (locationGroupById.containsKey(haltId)) {
-            return locationGroupById.get(haltId).location_group_name;
+        } else if (stopAreaById.containsKey(haltId)) {
+            return areaById.get(haltId).area_name;
```

Comment:
There has been a current issue identified with flex that if there is no stop area names, we will display "null" in the pattern names. Adding a condition here should resolve this. 
```suggestion
            Area area = areaById.get(haltId);
            return area.area_name != null ? area.area_name : area.area_id;
```